### PR TITLE
Resolving encoding issues

### PR DESCRIPTION
**Pull request by lsolodkova**
Will answer #51 and #52.

### CODE REVIEW

**Comment by mvondracek** [comment]
When this PR is ready for review, please change status from *draft* and assign reviewers so we get notification. :)

**Comment by lsolodkova** [comment]
Can't write a proper test for non-Unicode entropy/seed, exception occurs not while reading file, but only in unhexlify(). Any hints will be appreciated.

File: pa193mnemonicslytherin/mnemoniccli.py [hunk 1]
```diff
@@ -242,19 +242,32 @@ def action_generate(config: Config) -> ExitCode:
     try:
         with open(config.entropy_filepath, config.format.read_mode) as file:
             entropy = file.read()  # type: typing.Union[bytes, str]
-    except FileNotFoundError as e:
+    except (FileNotFoundError, UnicodeError) as e:
```
Comment (sobuch) [inline pa193mnemonicslytherin/mnemoniccli.py]: is there a reason for using this style and not something like the following?
```
try:
    ...
except error1:
   ...
except error2:
   ...
```

**Comment by sobuch** [comment]
please rebase on dev so we can fuzz this a little

File: pa193mnemonicslytherin/mnemoniccli.py (hunk 1 above)
Comment (lsolodkova) [reply pa193mnemonicslytherin/mnemoniccli.py]: No, I was hoping to reduce the amound of code for a bit, but this does not really matter. What to do with codeclimate issues don't know, I was trying not to modify your code much.

**Comment by lsolodkova** [comment]
Rebased on dev.

**Comment by lsolodkova** [comment]
Tests for non-Unicode entropy/seed work properly only if encoding is given explicitly in read() function (but I'm not able to make so cardinal changes to the code), otherwise we fail only inside unhexlify().

File: pa193mnemonicslytherin/mnemoniccli.py (hunk 1 above)
Comment (sobuch) [reply pa193mnemonicslytherin/mnemoniccli.py]: if you dont see some quick fixes for the complexity issues, we can probably let it be, I dont think its worth spending that much time on now (>8hrs estimated in codeclimate)

**Review by sobuch — DISMISSED** [review]

File: pa193mnemonicslytherin/mnemonic.py [hunk 2]
```diff
@@ -253,10 +259,10 @@ def to_seed(self, seed_password: str = '') -> Seed:
         if len(seed_password) > MAX_SEED_PASSWORD_LENGTH:
             raise ValueError('Password is too long')
         # the encoding of both inputs should be UTF-8 NFKD
-        mnemonic = self.encode()  # encoding string into bytes, UTF-8 by default
+        mnemonic = self.encode('utf-8')
         seed_password = normalize('NFKD', seed_password)
         passphrase = "mnemonic" + seed_password
-        passphrase = passphrase.encode()
+        passphrase = passphrase.encode('utf-8')
```
Comment (mvondracek) [inline pa193mnemonicslytherin/mnemonic.py:431]: `'utf-8'` is default value, so both version of this code are correct
https://docs.python.org/3/library/stdtypes.html#str.encode

**Review by mvondracek — COMMENTED** [review]
I've done refactoring and fixes. They are in separate branch. See #64

**Comment by mvondracek** [comment]
See #64 and #65

**Review by sobuch — APPROVED** [review]

**Review by mvondracek — APPROVED** [review]

**Pull request merged**